src/ray/rpc/grpc_client.h

```diff
@@ -88,6 +88,49 @@ inline std::shared_ptr<grpc::Channel> BuildChannel(
   return channel;
 }
 
+inline grpc::ChannelArguments CreateClientDefaultChannelArguments() {
+  // Please refer to explanation of gRPC returned statuses for more details:
+  // REF: https://grpc.github.io/grpc/core/md_doc_statuscodes.html
+  std::string service_config_json = R"(
+    {
+      "methodConfig": [{
+        "name": [
+          {"service": "ray.rpc.CoreWorkerService", "method": "ReportGeneratorItemReturns"},
+          {"service": "ray.rpc.CoreWorkerService", "method": "DeleteObjects"},
+          {"service": "ray.rpc.CoreWorkerService", "method": "CancelTask"},
+
+          {"service": "ray.rpc.CoreWorkerService", "method": "GetCoreWorkerStats"},
+          {"service": "ray.rpc.CoreWorkerService", "method": "NumPendingTasks"},
+          {"service": "ray.rpc.CoreWorkerService", "method": "PlasmaObjectReady"},
+          {"service": "ray.rpc.CoreWorkerService", "method": "GetObjectLocationsOwner"},
+          {"service": "ray.rpc.CoreWorkerService", "method": "GetObjectStatus"}
+        ],
+        "retryPolicy": {
+          "maxAttempts": 5,
+          "initialBackoff": "0.1s",
+          "maxBackoff": "1s",
+          "backoffMultiplier": 1.5,
+          "retryableStatusCodes": [
+              "UNAVAILABLE",
+              "DEADLINE_EXCEEDED",
+              "INTERNAL",
+              "UNKNOWN"
+          ]
+        }
+      }]
+    }
+  )";
+
+  grpc::ChannelArguments arguments = CreateDefaultChannelArguments(service_config_json);
```

Comment:
I'm not too familiar with grpc service config but I did a quick read and it seems it should be something set on the server side? The client side setting is mainly for unit tests?

Reply:
Nope, you also need to specify it on the client to configure retry policy